### PR TITLE
Secure the subscription passwords

### PR DESCRIPTION
**Pull request by RajuBeemreddy1**
### Applicable Issues

https://github.com/eiffel-community/eiffel-intelligence/issues/542

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:  <Raju Beemreddy> <raju.x.beemreddy@ericsson.com>

### CODE REVIEW

File: src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java [hunk 1]
```diff
@@ -97,11 +98,16 @@ private void extractConditions(String aggregatedObject,
         try {
             JsonNode subscriptionJson = new ObjectMapper().readTree(
                 subscriptionData);
-            LOGGER.debug("SubscriptionJson : " + subscriptionJson.toString());
-            LOGGER.debug("Aggregated Object : " + aggregatedObject  + " for the event id: " + id);
+			
+            // Remove password  from subscription details and put empty value before logging.
+			JsonNode subscriptoinToDisplay = new ObjectMapper().readTree(subscriptionJson.toString());
+			LOGGER.debug("SubscriptionJson : {}",
+					((ObjectNode) subscriptoinToDisplay).put("password", "").toPrettyString());
```
Comment (raja-maragani) [inline src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java:105]: Good to use the json ignore to skip secure fields.

**Review by raja-maragani — CHANGES_REQUESTED** [review]

File: src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java (hunk 1 above)
Comment (RajuBeemreddy1) [reply src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java:105]: As this is the json string, we can not use json ignore property here.

Comment (RajuBeemreddy1) [reply src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java:105]: I don't see any benefit in printing the whole subscription json in debug level as we are already printing the required info from the subscriptions in next line of debug statements. If we remove this, we can avoid this change/costly operation.  Please comment on this.

**Review by m-linner-ericsson — CHANGES_REQUESTED** [review]
I think we need a test on this. It's critical thing and if we don't have any test on it it could be that someone changes the code and the bug is back.

**Review by jainadc9 — APPROVED** [review]

**Pull request merged**